sdk/src/wallet/operations/block.rs

```diff
@@ -67,9 +71,20 @@ where
         self.emit(WalletEvent::TransactionProgress(TransactionProgressEvent::Broadcasting))
             .await;
 
+        log::debug!("submitting block {}", block.id(&protocol_parameters));
+        for attempt in 1..MAX_POST_BLOCK_ATTEMPTS {
+            if let Ok(block_id) = self.client().post_block(&block).await {
+                log::debug!("submitted block {}", block_id);
+                return Ok(block_id);
+            }
+            tokio::time::sleep(std::time::Duration::from_secs(attempt)).await;
+        }
+
+        log::debug!("submitting block {block:?}");
+
         let block_id = self.client().post_block(&block).await?;
```

Comment:
it's strange for this code to be duplicated just for the return value. Is there no error for running out of attempts?

Reply:
Currently we don't have one, ofc we could add one, but I don't think it's really worth it

Reply:
any way to remove the duplicated code then?

Reply:
Yes, but not sure if that's really better now

Reply:
I think it is, because it returns the error